### PR TITLE
enh(http) support headers without preamble

### LINKED ISSUE #3019: (http) highlight http headers without HTTP preamble
**Describe the issue**
It should be possible to declare http key/value pairs without requiring HTTP preamble.  

For example, just:

```HTTP
Some-header: value
```

**Which language seems to have the issue?**
HTTP

**Are you using `highlight` or `highlightAuto`?**
highlightAuto

...

**Sample Code to Reproduce**

```HTTP
Referer: value
```

**Expected behavior**

Should output:

```HTML
<span class="hljs-attribute">Referer</span>
```

**Additional context**
Appears to be a regression from 10.5. It used to work. Maybe #2893?

### PR DESCRIPTION
**Pull request by joshgoebel**
Resolves #3019.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [ ] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by joshgoebel** [comment]
@marcoscaceres

File: test/markup/cpp/function-declarations.expect.txt [hunk 1]
```diff
@@ -21,4 +21,3 @@
 
 <span class="hljs-function"><span class="hljs-keyword">void</span> <span class="hljs-title">A</span><span class="hljs-params">()</span>: a(<span class="hljs-number">10</span>) {</span>}
 <span class="hljs-function"><span class="hljs-keyword">explicit</span> <span class="hljs-title">A</span><span class="hljs-params">()</span>: a(<span class="hljs-number">10</span>) {</span>}
```
Comment (allejo) [inline test/markup/cpp/function-declarations.expect.txt:23]:
```suggestion
<span class="hljs-function"><span class="hljs-keyword">explicit</span> <span class="hljs-title">A</span><span class="hljs-params">()</span>: a(<span class="hljs-number">10</span>) {</span>}

```

The C++ markup expected file is being touched in this PR by accident it seems. Not sure if you'd want to squash/clean it up somehow.

**Review by allejo — APPROVED** [review]

File: test/markup/cpp/function-declarations.expect.txt (hunk 1 above)
Comment (joshgoebel) [reply test/markup/cpp/function-declarations.expect.txt:23]: This is just fixing a line break issue, so I don't really care.

**Comment by marcoscaceres** [comment]
Thanks for working on this, @joshgoebel!

**Pull request merged**